pkg/webhook/validation.go

```diff
@@ -102,8 +102,8 @@ func (v *validator) validate(ctx context.Context, obj runtime.Object) (admission
 			sourceCount++
 			unionCount++
 
-			if len(configMap.Name) == 0 {
-				el = append(el, field.Invalid(path.Child("name"), configMap.Name, "source configMap name must be defined"))
+			if len(configMap.Name) == 0 && configMap.Selector == nil {
```

Comment:
We must also validate that the user hasn't set both `name` and `selector`.

Reply:
I just added a test for that. Let me know what you think!

Reply:
Looks like a good start, but a bit more work is needed. We could see if there is a way to patch the CRD with a [oneOf](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/), which is supported by OpenAPI/Kubernetes, but not yet by c/r, ref. https://github.com/kubernetes-sigs/controller-tools/issues/461.

Reply:
From what I have found Googling around a bit, most people seems to manually add that validation into a webhook instead of patching the CRD afterwards (because it gets overwritten every time we re-generate the CRDs).

See my other comments on the other threads about this issue: https://github.com/cert-manager/trust-manager/pull/258#discussion_r1445222407

Reply:
We already have a webhook, so I think there is no need for a oneOf OpenAPI validation.